addons/web_editor/static/src/components/media_dialog/media_dialog.js

```diff
@@ -184,6 +184,11 @@ export class MediaDialog extends Component {
                         if (this.props.media.dataset.shapeColors) {
                             element.dataset.shapeColors = this.props.media.dataset.shapeColors;
                         }
+                    } else if ([TABS.VIDEOS.id, TABS.DOCUMENTS.id].includes(this.state.activeTab)) {
+                        const parentEl = this.props.media.parentElement;
+                        if (parentEl.tagName === "A" && parentEl.children.length === 1 && this.props.media.tagName === "IMG") {
```

Comment:
Add a comment here, for instance: "If an image is wrapped in an `<a>` tag, we remove the link when replacing it with a video or document".